### ISSUE TITLE
vsce package on marketplace different than when building a package

### ISSUE THREAD

**Issue by juli1** [microsoft/vscode-vsce#583]
Hello,

When I build my package using `vsce package`, it produces a file I can use and install manually in vscode. Everything works.

When I use `vsce publish`, it uploads a file on the marketplace that has a contents different than the package I have with `vsce package`. In particular, the `extension.js` file (the main entrypoint) is replaced into a file called `main.js`. As a result, the main entrypoint is not found when trying to start the extension. So the extension works when built locally (using `vsce package`) but the extension on the marketplace does not work.

Code can be found here: https://github.com/codeinspectorio/vscode-plugin

The package on the marketplace (not working) can be found here: https://marketplace.visualstudio.com/items?itemName=code-inspector.code-inspector-vscode-plugin&ssr=false#overview

Not sure if this is a problem related to vsce, my configuration or the marketplace itself. If not the right place and you know where is the right place, please let me know.

Thanks!

**Comment by joaomoreno** [comment]
> In particular, the `extension.js` file (the main entrypoint) is replaced into a file called `main.js`.

Can you show me this? I don't see that when I download the VSIX and look at it myself:

![image](https://user-images.githubusercontent.com/22350/122556809-cb02ef80-cff0-11eb-94e9-11a9d26681dc.png)

**Comment by juli1** [comment]
Hi @joaomoreno, sorry for keeping this ticket opened, I found the issue and the new package is fixed.

What happened is the following: if I sent a package to the marketplace, the content with the package.js was replaced with a main.js. To see what happened, I unzipped the package, renamed main.js to package.js, rezip the file and load it in VS Code manually. It happened that the plugin was not loading because of two different versions of `graphql` in the dependencies ([an issue found here](https://stackoverflow.com/questions/59272819/cannot-use-e-schema-from-another-module-or-realm-and-duplicate-graphql) by others when using `apollo-client`).

To fix the issue, I removed my dependency on `apollo-client` (used the one from axios) and everything is now working.

I suspect that during the verification process, the content of the code is analyzed, maybe repackaged, and the tool failed and just produced an artifact (`main.js`). Of course, that is just an assumption since I have no visibility on the verification process on the VS Code marketplace.

I am closing this ticket because (1) it seems related to the marketplace website and not directly to `vsce` and (2) I no longer have this problem. Sorry for not having closed it earlier.
